corehq/apps/zapier/tests.py

```diff
@@ -329,3 +350,46 @@ def test_form(self):
         self.assertIsNone(form.get('partial_submission'))
 
         self.assertIsNotNone(form['domain'])
+
+
+class TestZapierCustomFields(TestCase):
+
+    def test_case_fields(self):
+
+        expected_fields = [
+            {"help_text": "", "key": "properties__level", "label": "Level", "type": "unicode"},
+            {"help_text": "", "key": "properties__mood", "label": "Mood", "type": "unicode"},
+            {"help_text": "", "key": "properties__move", "label": "Move", "type": "unicode"},
+            {"help_text": "", "key": "properties__name", "label": "Name", "type": "unicode"},
+            {"help_text": "", "key": "properties__opened_on", "label": "Opened on", "type": "unicode"},
+            {"help_text": "", "key": "properties__owner_id", "label": "Owner id", "type": "unicode"},
+            {"help_text": "", "key": "properties__prop1", "label": "Prop1", "type": "unicode"},
+            {"help_text": "", "key": "properties__type", "label": "Type", "type": "unicode"},
+            {"help_text": "", "key": "date_closed", "label": "Date closed", "type": "unicode"},
+            {"help_text": "", "key": "xform_ids", "label": "XForm IDs", "type": "unicode"},
+            {"help_text": "", "key": "properties__date_opened", "label": "Date opened", "type": "unicode"},
+            {"help_text": "", "key": "properties__external_id", "label": "External ID", "type": "unicode"},
+            {"help_text": "", "key": "properties__case_name", "label": "Case name", "type": "unicode"},
+            {"help_text": "", "key": "properties__case_type", "label": "Case type", "type": "unicode"},
+            {"help_text": "", "key": "user_id", "label": "User ID", "type": "unicode"},
+            {"help_text": "", "key": "date_modified", "label": "Date modified", "type": "unicode"},
+            {"help_text": "", "key": "case_id", "label": "Case ID", "type": "unicode"},
+            {"help_text": "", "key": "properties__owner_id", "label": "Owner ID", "type": "unicode"},
+            {"help_text": "", "key": "resource_uri", "label": "Resource URI", "type": "unicode"}
+        ]
+
+        request = Client().get(TEST_URL).wsgi_request
+        bundle = Resource().build_bundle(data={}, request=request)
+
+        factory = CaseFactory(domain="johto")
+        factory.create_case(
+            case_type='teddiursa',
+            owner_id='owner1',
+            case_name='dre',
+            update={'prop1': 'blah', 'move': 'scratch', 'mood': 'happy', 'level': '100'}
```

Comment:
you could also test a property which you expect to have a whitespace character substituted eg: `'pokedex_colour': 'brown'`